pkg/service/legacy/service_test.go

```diff
@@ -0,0 +1,252 @@
+		indexes := bytemap.NewByteMap[types.EncodedContextID, blobindex.ShardedDagIndexView](0)
+		result, err := queryresult.Build(claims, indexes)
+		require.NoError(t, err)
+
+		mockService := mockIndexingService{result, nil}
+		service := NewService(id, &mockService, mockStore, *bucketURL)
+		query := types.Query{Hashes: []multihash.Multihash{digest}}
+
+		results, err := service.Query(context.Background(), query)
+		require.NoError(t, err)
+		require.Empty(t, results.Indexes())
+		require.Equal(t, 1, len(results.Claims()))
+		require.Equal(t, claim.Link(), results.Claims()[0])
+	})
+
+	t.Run("returns indexes from underlying indexing service", func(t *testing.T) {
+		mockStore := newMockBlockIndexStore()
+		root := testutil.RandomCID()
+		digest := link.ToCID(root).Hash()
+
+		index := blobindex.NewShardedDagIndexView(root, 0)
+		indexBytes, err := io.ReadAll(testutil.Must(index.Archive())(t))
+		require.NoError(t, err)
+
+		indexCID, err := cid.Prefix{
+			Version:  1,
+			Codec:    uint64(multicodec.Car),
+			MhType:   multihash.SHA2_256,
+			MhLength: -1,
+		}.Sum(indexBytes)
+		require.NoError(t, err)
+
+		claims := map[cid.Cid]delegation.Delegation{}
+		indexes := bytemap.NewByteMap[types.EncodedContextID, blobindex.ShardedDagIndexView](1)
+		indexes.Set(types.EncodedContextID(digest), index)
+		result, err := queryresult.Build(claims, indexes)
+		require.NoError(t, err)
+
+		mockService := mockIndexingService{result, nil}
+		service := NewService(id, &mockService, mockStore, *bucketURL)
+		query := types.Query{Hashes: []multihash.Multihash{digest}}
+
+		results, err := service.Query(context.Background(), query)
+		require.NoError(t, err)
+		require.Empty(t, results.Claims())
+		require.Equal(t, 1, len(results.Indexes()))
+		require.Equal(t, indexCID.String(), results.Indexes()[0].String())
+	})
+
+	t.Run("calls through to underlying indexing service", func(t *testing.T) {
+		digest := testutil.RandomMultihash()
+		mockStore := newMockBlockIndexStore()
+		mockService := mockIndexingService{nil, errNotImplemented}
+		service := NewService(id, &mockService, mockStore, *bucketURL)
+
+		nb := assert.LocationCaveats{Content: assert.FromHash(digest)}
+		claim, err := assert.Location.Delegate(id, id, id.DID().String(), nb)
+		require.NoError(t, err)
+
+		err = service.Cache(context.Background(), peer.AddrInfo{}, claim)
+		require.True(t, errors.Is(err, errNotImplemented))
+
+		_, err = service.Get(context.Background(), testutil.RandomCID())
+		require.True(t, errors.Is(err, errNotImplemented))
+
+		err = service.Publish(context.Background(), claim)
+		require.True(t, errors.Is(err, errNotImplemented))
+
+		query := types.Query{Hashes: []multihash.Multihash{digest}}
+		_, err = service.Query(context.Background(), query)
+		require.True(t, errors.Is(err, errNotImplemented))
+	})
+}
+
+var errNotImplemented = errors.New("not implemented")
+
+type mockIndexingService struct {
+	queryResult types.QueryResult
+	queryError  error
+}
```

Comment:
alternatively, you could get testify-compatible mocks created automatically using `mockery` and express expectations in tests directly. Take a look at the tests in my PR #51 or ping me if you feel like going that route.

Reply:
These are so simple I feel like it might be overkill but maybe you could show me how to generate?

Reply:
let's take a look at it in the next demo session, as you suggested 👍🏻